File: app/frontend/modules/survey/models/Validations/Most.js
import _ from 'lodash'
import { I18n } from 'modules/survey/store/StoreWatchman'

const Most = function ({ maxValue }) {
  this.maxValue = +maxValue
}

_.extend(Most.prototype, {
  validate (answers) {
    if (answers.length > this.maxValue) {
      return {
        type: 'Most',
        message: I18n().t('validations.most', { max: this.maxValue }),
      }
    }
  },
})

export default Most

File: app/core/questions/validations/most.rb
# frozen_string_literal: true

module Questions::Validations
  class Most < BaseCommand
    attr_accessor :args, :message, :value

    def initialize(validation, value, locale)
      @args = validation['args']
      @value = value || []
      @locale = locale
    end

    def call
      return broadcast :ok, nil unless args['maxValue']

      invalid = value.length > args['maxValue']

      broadcast :ok, invalid ? [I18n.t('validations.most', max: args['maxValue'])] : nil
    end
  end
end
